### PR TITLE
Modal Size and fix modal touch events

### LINKED ISSUE #3127: [Bug] View underneath Modal Page is currently focusable by keyboard and TalkBack
Proposed Fix is to add this to ModalNavigationManager

```C#
			if (_renderer.ChildCount > 0 && _renderer.GetChildAt(0) is AView view)
			{
				view.ImportantForAccessibility = ImportantForAccessibility.NoHideDescendants;

				if (NativeVersion.IsAtLeast(26))
					view.SetFocusable(ViewFocusability.NotFocusable);

				if (view is ViewGroup vg)
					vg.DescendantFocusability = DescendantFocusability.BlockDescendants;
			}
```

Validate UpdateTablayoutVisibility() in ShellSectionView to see if the ViewPager there also needs to be modified so it doesn't gain focus

### LINKED ISSUE #5627: Shell modal page triggers TapGestureRecognizer which lies on the page under it.
### Description

I have a shell page that contains a tab layout with one page on which I use a TapGestureRecognizer that throws an exception on tap. I placed a button that shows the modal window. If you click on the modal window on a green rectangle (this area has TapGestureRecognizer on the previous page) then the application will crash because it triggers a tap event for the previous page.

### Steps to Reproduce

I created a project to replicate an issue and placed it [here](https://github.com/nevse/Maui-Tests-ShellExampleGestureR-).

Click on the button "Open Modal" then click the green rectangle on the opened modal page. App will crash because the TapGestureRecognizer was triggered from the previous page (it throw an exception).

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11

### Did you find any workaround?

I didn't find any workaround.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

- When the user touches anywhere on the Modal that's not an input control we need to set the touch as handled so it doesn't pass through to the view underneath
- Measure the Modal window to match the same position/size as the underlying view. This means we need to take into account the statusbar and the navigationbar

### Issues Fixed

Fixes #5627
Fixes #4961

### Accessibility

Validate that accessibility/keyboard navigation still works as expected when you push a modal page. If this is all working correctly we can close https://github.com/dotnet/maui/issues/3127

### CODE REVIEW

**Review by rachelkang — APPROVED** [review]
LGTM!

**Review by jsuarezruiz — APPROVED** [review]

File: src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs [hunk 1]
```diff
@@ -211,6 +211,9 @@ sealed class ModalContainer : ViewGroup
 				UpdateBackgroundColor();
 				AddView(_backgroundView);
 
+				// Don't let touch events pass through to the view being covered up
+				this.Touch += (_, args) => args.Handled = true;
```
Comment (Redth) [inline src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs:215]: Are we good not unregistering this?

Comment (PureWeen) [reply src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs:215]: The only reason to unregister is if subscribing to the platform events will cause memory leaks or other types of collection issues/delays.  These views are all single use for each modal that pops up, so this subscription isn't going to get hit multiple times and the platform modals container isn't reused.

Comment (PureWeen) [reply src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs]: I switched this code over to just use `override TouchEvent`

**Review by Redth — APPROVED** [review]

**Comment by AmrAlSayed0** [comment]
There is a chance that the status bar would be disabled/hidden (like it is in our apps) or the orientation of the modal page is landscape (I don't know if that switches the l, t, r, b values)

**Pull request merged**